### ISSUE TITLE
Feature Request: Open-Open in Zotero

### ISSUE THREAD

**Issue by kzssc** [chrisgrieser/alfred-bibtex-citation-picker#44]
### Feature Requested

Current behaviour with "Open in Zotero" (triggered by shift when an entry is selected) is that it locates the entry within zotero then highlights it. 

(One issue (not of the plugin, I think) here is that it does not work when Zotero is closed. When this is executed w/ Zotero closed, Zotero merely launches *without* finding/highlighting the entry. One needs to execute the execute command again to trigger the highlight from within Zotero.)

I propose a feature that actually opens the entry/file in zotero, in addition to locating this entry. Optionally and better: if the opening can be done external to zotero. (e.g. an external PDF viewer.) 

The reason to this: open-in-zotero could take some time for Zotero, althought it does work. 

### Checklist

- [X] The feature would be useful to more users than just me.

**Comment by chrisgrieser** [comment]
Makes sense. However, I personally do not use Zotero, so I am not really familiar with the internals of how it works.

Currently this workflow opens the entry by calling the URI scheme `zotero://select/items/bbt:{CITEKEY}`. Do you (or anyone else) know of a URI scheme that does open the file like you described? Something like that would be required to be able to implement such a feature.

**Comment by kzssc** [comment]
> Makes sense. However, I personally do not use Zotero, so I am not really familiar with the internals of how it works.
> 
> Currently this workflow opens the entry by calling the URI scheme `zotero://select/items/bbt:{CITEKEY}`. Do you (or anyone else) know of a URI scheme that does open the file like you described? Something like that would be required to be able to implement such a feature.

Thanks for the immediate reply, Chris. 

Personally i don't know of anything/anyone that has a better way. (I vaguely remember [this](https://github.com/deanishe/zothero) and [the fork](https://github.com/giovannicoppola/zothero) can do this but it generates a new temp file etc... i think it's too little to gain for a massive effort and trouble, so no.)

Nonetheless, as the .bib contains attachment directory, so possibly, a trigger might point at that?

**Comment by chrisgrieser** [comment]
could you share one or two example entries with a link to an attachment?

**Comment by kzssc** [comment]
sure, here! https://transfer.archivete.am/SaIWC/2%20Lottery.bib

**Comment by chrisgrieser** [comment]
okay, added the ability to open attachment files directly via `ctrl+alt+cmd+return` (sorry, there aren't enough modifiers left). Attachment files are files that have the key "file" or "attachment" in the BibTeX entry.

The Alfred gallery will auto-update within ~2-3 days, or you can simply download the new version directly:
https://github.com/chrisgrieser/alfred-bibtex-citation-picker/releases/download/11.14.0/alfred-bibtex-citation-picker.alfredworkflow

**Comment by daaaaaaaaaniel** [comment]
It's not working for me. I set my PDF folder to the location where Zotero stores PDF `~/Zotero/storage/` but alas. I get the disappointed sound when I try to open the file in Zotero. The `zotero://select/items/` version succeeds by holding shift, but when I add the extra modifiers to open the PDF, nothing.

**Comment by chrisgrieser** [comment]
@daaaaaaaaaniel The current implementation is based on how the entries from `kzssc` look like. Likely, your entries are formatted differently. 

Could you share an example of how your entries look like?

**Comment by kzssc** [comment]
> Makes sense. However, I personally do not use Zotero, so I am not really familiar with the internals of how it works.
> 
> Currently this workflow opens the entry by calling the URI scheme `zotero://select/items/bbt:{CITEKEY}`. Do you (or anyone else) know of a URI scheme that does open the file like you described? Something like that would be required to be able to implement such a feature.

from https://github.com/giovannicoppola/zothero/issues/30 and https://forums.zotero.org/discussion/90858/pdf-reader-and-zotero-open-pdf-links it seem that with "zotero: //open-pdf"  instead of the current "zotero://select" we can open the pdf in zotero (?)

**Comment by chrisgrieser** [comment]
> from giovannicoppola/zothero#30 and https://forums.zotero.org/discussion/90858/pdf-reader-and-zotero-open-pdf-links it seem that with "zotero: //open-pdf" instead of the current "zotero://select" we can open the pdf in zotero (?)

if it is about opening attachments/PDFs, the current approach should already work, and has also the benefit for working for people not using Zotero. So I don't see the benefit of switching to `zotero: //open-pdf`, or am I missing something?

Otherwise, `zotero://select` is for revealing the entry in Zotero, which, to my knowledge, also works.

If you have a different feature suggestions, it's probably better to open a new FR where you outline what exactly you are thinking of?
